core/src/main/java/org/apache/carbondata/scan/executor/impl/AbstractQueryExecutor.java

```diff
@@ -470,7 +471,12 @@ private AggregatorInfo getAggregatorInfoForBlock(QueryModel queryModel,
    */
   @Override public void finish() throws QueryExecutionException {
     if (null != queryProperties.executorService) {
-      queryProperties.executorService.shutdownNow();
+      queryProperties.executorService.shutdown();
+      try {
+        queryProperties.executorService.awaitTermination(1, TimeUnit.HOURS);
```

Comment:
This is called when closing the reader for every query, I think timeout for 1 hour is too long. Suggest to change to 10 seconds.